src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs

```diff
@@ -31,14 +33,15 @@ protected internal override void Setup(FeatureConfigurationContext context)
             }
 
             var conventions = context.Settings.Get<Conventions>();
+            var transportInfrastructure = context.Settings.Get<TransportInfrastructure>();
+            var requireExplicitRouting = transportInfrastructure.OutboundRoutingPolicy.Publishes == OutboundRoutingType.Unicast;
+            var publishers = context.Settings.Get<Publishers>();
 
             context.RegisterStartupTask(b =>
             {
                 var handlerRegistry = b.Build<MessageHandlerRegistry>();
-
                 var messageTypesHandled = GetMessageTypesHandledByThisEndpoint(handlerRegistry, conventions, settings);
-
-                return new ApplySubscriptions(messageTypesHandled, type => TaskEx.TrueTask);
+                return new ApplySubscriptions(messageTypesHandled, eventType => !requireExplicitRouting || publishers.GetPublisherFor(eventType).Any());
```

Comment:
couldn't we apply the predicate just here and simplify `ApplySubscriptions` a lot?